### ISSUE TITLE
tar: .BUILDINFO: Cannot open: File exists

### ISSUE THREAD

**Issue by BenSzanta** [drinkcat/chroagh#104]
Hi, I've got a problem with installing. When I start the script it ended here:

> chronos@localhost ~/Downloads/drinkcat-chroagh-18cc180 $ sudo sh -e installer/main.sh -r arch -t core,audio,extension,keyboard,xfce-desktop,
> WARNING: USB booting is enabled; consider disabling it.
> WARNING: Legacy booting is enabled; consider disabling it.
> WARNING: Signed boot verification is disabled; consider enabling it.
> You can use the following command: sudo crossystem dev_boot_usb=0 dev_boot_legacy=0 dev_boot_signed_only=1
> Installing arch-x86_64 chroot to /usr/local/chroots/arch
> Fetching repository packages list...
> Fetching core...
> ###### ################################################################## 100.0%
> 
> .
> .
> .
> Downloading and installing nettle...
> ###### ################################################################## 100.0%
> 
> tar: .BUILDINFO: Cannot open: File exists
> tar: Exiting with failure status due to previous errors
> chronos@localhost ~/Downloads/drinkcat-chroagh-18cc180 $ 

I think the problem is that .BUILDINFO already exists, but I don't know it's location so I can't delete it.
Thanks for the help

**Comment by BenSzanta** [comment]
Allright. I fix it same way as libcursesw.so. 
Just by adding:

> > if [ "$package" = "nettle" ]; then 
> >     echo "skipping $package"
> >     installed="$installed $package "
> >     fi;
> > to the bootstrap file in /installer/arch/

I don't know what was wrong with it but this fixes it.

**Comment by bendavis78** [comment]
Same issue here, thanks for the fix.

**Comment by stevendlander** [comment]
You can also fix this by giving the tar command in `installers/arch/bootstrap' the`--overwrite` flag.  This should get the file copy moving and clobber any lingering .BUILDINFO files of any packages instead of skipping them entirely.

I will put this in a PR and then link to your issue so you can see/download the file for future use.

**Comment by Arthurwu24** [comment]
I got the same problem, but a little bit different,although the same result in the shell...

tar: grep-2.24-1/depends: Not found in archive
tar: Exiting with failure status due to previous errors

And i don't know how to fix this.. Coz  i can't boot into the arch.

PLZ:   I have add --overwrite in the bootrap ...  No good.

**Comment by stevendlander** [comment]
@Arthurwu24 That is a separate error from the `File exists` error this issue reports.  I would suggest making a new issue for that, since this issue was handled by #107 already and _should_ be closed by @drinkcat.

That error is probably due to variable expansion happening in the bootstrap script, [like this link suggests.](http://stackoverflow.com/questions/16933221/shell-tar-not-found-in-archive-error-when-using-regular-expression)

**Comment by Arthurwu24** [comment]
@stevendlander So what shall i do to deal with it?

**Comment by stevendlander** [comment]
@Arthurwu24 Short term: you can try a different fork of this project since @drinkcat might have abandoned the repo. @elkangaroo just recently merged my proposed change to his chroargh repo so they may have already fixed this issue.  Try using [their repo](https://github.com/elkangaroo/chroagh) to see if the issue persists.

Long term, the bootstrap.sh file needs to be debugged to figure if multiple command-line arguments are getting expanded and passed to tar or not.  Tar is being called only in a few places in the script so it could be a trivial fix if you looked at it.  I wouldn't mind fixing but I have no idea when I would get some free cycles to look.  :(

**Comment by elkangaroo** [comment]
@Arthurwu24 Yes, you can try using my repo as @stevendlander proposed. I just recently rebased with crouton master, added different fixes for the installer and fixed some remaining merge issues.

So far, everything installs just fine (on Acer C720 Chromebook) and all the non-graphical targets seem to work great, but I did not get X.org to work yet, and hence none of the desktop environments.

**Comment by Arthurwu24** [comment]
@stevendlander Thx,bro....But i tried the repo from @elkangaroo , still the same issue......My device is Acer chromebook cb3-11 ..I still stucked in the problem like i posted before :

tar: grep-2.24-1/depends: Not found in archive
tar: Exiting with failure status due to previous errors

And i can't boot into Arch due to this... What else can i try ?

**Comment by Arthurwu24** [comment]
@elkangaroo ..I'm sorry,pal. I just found the command line in your chroagh page is the same as drinkcat's.so maybe i got something wrong? Coz that means i typed the same command line in shell...so  its fair that i got the same issue .... so what should i do,sorry for bother u so much..

**Comment by elkangaroo** [comment]
@Arthurwu24 did you use the master branch from my fork? Because the error you get should be fixed there with commit 7b5720c0f0d0f4e2a800b1cf216369843c4061da.

What is the exact command you are running from your shell?

**Comment by Arthurwu24** [comment]
@elkangaroo . I'm sorryl, i don't get it.. What's the meaning of using master branch from your fork??? I just typed the command : 

cd ~/Downloads
wget https://api.github.com/repos/drinkcat/chroagh/tarball -O chroagh.tar.gz
tar xvf chroagh.tar.gz
cd drinkcat-chroagh-*

sudo sh -e installer/main.sh -r arch -t xfce

I guess i did sth wrong.. Cause this is the command showed in drinkcat/chroagh..... But when i use your repo..I typed the same command.... What exact command shall i type ??

**Comment by NoahVlncrt** [comment]
@Arthurwu24 Just tested his repo out. You want to replace drinkcat with @elkangaroo

**Comment by baffometo** [comment]
@elkangaroo man i wanted to say thanks for some reason your repo works. I was checking the codes but i dont understand why ur works lol do u mind to tell how u did it?
Thanks man! finally after hours of working i have my archlinux Up and running!

**Comment by jayala-wt** [comment]
@elkangaroo Hi there i used ur repo but i have this error
 patch-2.7.5-1-i686                                               87.7 KiB   669K/s 00:00 [####################################################] 100%
(1/1) checking keys in keyring                                                            [####################################################] 100%
(1/1) checking package integrity                                                          [####################################################] 100%
(1/1) loading package files                                                               [####################################################] 100%
(1/1) checking for file conflicts                                                         [####################################################] 100%
(1/1) checking available disk space                                                       [####################################################] 100%
:: Processing package changes...
(1/1) installing patch                                                                    [####################################################] 100%
Optional dependencies for patch
    ed: for patch -e functionality
Compiling CRAS (native)...
common/cras_config.c: In function 'cras_config_get_system_socket_file_dir':
common/cras_config.c:16:9: error: 'CRAS_SOCKET_FILE_DIR' undeclared (first use in this function)
  return CRAS_SOCKET_FILE_DIR;
         ^~~~~~~~~~~~~~~~~~~~
common/cras_config.c:16:9: note: each undeclared identifier is reported only once for each function it appears in
common/cras_config.c:17:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
Failed to complete chroot setup.
Unmounting /mnt/stateful_partition/crouton/chroots/arch...

**Comment by jayala-wt** [comment]
@elkangaroo and when i type startxfce4 i have the following info

chronos@localhost ~/Downloads $ sudo startxfce4
Entering /mnt/stateful_partition/crouton/chroots/arch...
A chroot setup script still exists inside the chroot.
The chroot may not be fully set up.
Would you like to finish the setup? [Y/n/d] y
Preparing chroot environment...
chown: invalid user: 'messagebus:messagebus'
Installing target core...
Preparing environment...
Ensuring system is up-to-date...
:: Synchronizing package databases...
 core is up to date                       0.0   B  0.00B/s 00:00 [-----------------------------------]   0%
 extra is up to date                      0.0   B  0.00B/s 00:00 [-----------------------------------]   0%
 community is up to date                  0.0   B  0.00B/s 00:00 [-----------------------------------]   0%
:: Starting full system upgrade...
 there is nothing to do
warning: sudo-1.8.17.p1-1 is up to date -- skipping
warning: wget-1.18-1 is up to date -- skipping
warning: ca-certificates-20160507-1 is up to date -- skipping
 there is nothing to do
Syncing timezone...
Installing target audio...
Fetching CRAS (branch 0dac4acd30b69f02498ca4696f5660800ec848be)...
--2016-08-28 22:37:19--  https://chromium.googlesource.com/chromiumos/third_party/adhd/+archive/0dac4acd30b69f02498ca4696f5660800ec848be.tar.gz
Resolving chromium.googlesource.com (chromium.googlesource.com)... 2607:f8b0:400e:c04::52, 74.125.28.82
Connecting to chromium.googlesource.com (chromium.googlesource.com)|2607:f8b0:400e:c04::52|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/x-gzip]
Saving to: '/tmp/crouton-cras.740zQm/adhd.tar.gz'

```
 0K .......... .......... .......... .......... ..........  427K
```

   500K .......... .......... .......... .......... .....      1.55M=0.4s

2016-08-28 22:37:20 (1.24 MB/s) - '/tmp/crouton-cras.740zQm/adhd.tar.gz' saved [558096]

warning: alsa-utils-1.1.2-1 is up to date -- skipping
warning: alsa-lib-1.1.2-1 is up to date -- skipping
warning: speex-1.2rc2-1 is up to date -- skipping
 there is nothing to do
warning: gcc-6.1.1-5 is up to date -- skipping
warning: pkg-config-0.29.1-2 is up to date -- skipping
 there is nothing to do
warning: patch-2.7.5-1 is up to date -- skipping
 there is nothing to do
Compiling CRAS (native)...
common/cras_config.c: In function 'cras_config_get_system_socket_file_dir':
common/cras_config.c:16:9: error: 'CRAS_SOCKET_FILE_DIR' undeclared (first use in this function)
  return CRAS_SOCKET_FILE_DIR;
         ^~~~~~~~~~~~~~~~~~~~
common/cras_config.c:16:9: note: each undeclared identifier is reported only once for each function it appears in
common/cras_config.c:17:1: error: control reaches end of non-void function [-Werror=return-type]
 }
 ^
cc1: all warnings being treated as errors
Failed to complete chroot setup.
The chroot setup script may be broken. Your chroot is not fully configured.
Removing the chroot setup script. You may want to update your chroot again.
UID 1000 not found in arch
Unmounting /mnt/stateful_partition/crouton/chroots/arch...

**Comment by elkangaroo** [comment]
Hi @baffometo thx for reporting this. The error is happening due to a upstream change (see issue https://github.com/dnschneid/crouton/issues/2583 in the crouton repository) and was fixed in pull request https://github.com/dnschneid/crouton/pull/2589.

I will attempt a rebase of dnschneid/crouton master either today or tomorrow to get the fix into https://github.com/elkangaroo/chroagh.

**Comment by jayala-wt** [comment]
@elkangaroo thanks you are amazing

**Comment by deahl1** [comment]
I get this error when I run the main.sh script:

tar: '--overwrite' cannot be used with '--keep-old-files'

I was getting the .BUILDINFO error, and used this repo instead...
Please help!!
